### PR TITLE
Added support for offset, scrollTop and minScrollHeight to support nested search-bar (closes #38)

### LINKED ISSUE #38: Support a classic search field as part of the list
For the "raw" `fast-list.js` it's easy
- you put your search field in the dom
- style + add some snapping, since the list uses "native" scrolling it's easy to use css snapping to prevent the field from being "half-openned"
- you offset everything by the height of the field in the datasource (positions + total length)

For the component we'll need
- a config flag for this "padding"
- to move the field's DOM (inside the component) to the container? Not sure about this one...

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by wilsonpage** [comment]
Based on top of #48

**Comment by etiennesegonzac** [comment]
This one can be rebased now :)

File: examples/gaia-fast-list/offsets/app.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+var list = document.querySelector('gaia-fast-list');
+
+list.scrollTop = 50;
+list.minScrollHeight = 'calc(100% + 50px)';
+list.model = getData();
+
+function getData() {
+  var result = [];
+
+  for (var i = 0; i < 1000; i++) {
```
Comment (etiennesegonzac) [inline examples/gaia-fast-list/offsets/app.js]: should we make this smaller in order to demo the `minScrollHeight` part?

**Comment by etiennesegonzac** [comment]
I'll give this one a last quick look once rebased to make sure I didn't miss anything, but it's good!

File: gaia-fast-list.js [hunk 2]
```diff
@@ -283,10 +312,15 @@ Internal.prototype = {
   createList: function() {
     debug('create list');
     this.fastList = new FastList(this);
-    this.el.classList.add('layerize');
+    setTimeout(() => this.layerize(), 360);
```
Comment (etiennesegonzac) [inline gaia-fast-list.js:315]: not sure what will end up being the best performing solution here, but let's land this and follow up on it :)

**Pull request merged**